pkgs/build-support/fetchurl/default.nix

```
@@ -117,6 +117,9 @@ let
     else throw "fetchurl requires either `url` or `urls` to be set";
 
   hash_ =
+    # Many other combinations don't make sense, but this is the most common one:
+    if hash != "" && sha256 != "" then throw "multiple hashes passed to fetchurl" else
```

```suggestion
```suggestion
    if hash != "" && sha1 != "" && sha256 != "" && sha512 != "" then throw "multiple hashes passed to fetchurl" else
```
might as well. Also what about we drop sha1 like md5?
```

---

That's not how you check for existance of more than one hash though.

---

You suggest wrong conditions.

---

That will only fail on all of them present, so it would have to be a `hash & sha1 || hash & sha256 || hash & sha512 || sha1 & sha256....` or some kind of `(count-non-empty-in [hash sha1 sha256 sha512]) > 1

---

Before outright dropping the support I'd go through some time when it just prints an annoying warning.  And I haven't checked the current usage in nixpkgs and tools.  Anyway, that's way bigger than this PR and there's an issue already: https://github.com/NixOS/nixpkgs/issues/77238